src/fhirtypes/CodeSystem.ts

```diff
@@ -2,13 +2,16 @@ import { Meta } from './specialTypes';
 import { Extension } from '../fshtypes';
 import { Narrative, Resource, Identifier, CodeableConcept, Coding } from './dataTypes';
 import { ContactDetail, UsageContext } from './metaDataTypes';
+import { FHIRId } from './primitiveTypes';
+import { HasName, HasId } from './common';
+import { applyMixins } from '../utils';
 
 /**
  * Class representing a FHIR R4 CodeSystem
  * @see {@link https://www.hl7.org/fhir/codesystem.html}
  */
 export class CodeSystem {
-  id?: string;
+  id?: FHIRId;
```

Comment:
My understanding is that we no longer need to declare this property as it gets mixed in, right?  Looking at the documentation you linked to, the base class does not declare the properties that are already defined in the mixed in classes.

One potential issue w/ leaving this here is if we change the property here or in the mixin (but not in the other), then I'm not sure what happens.  That said, if we remove it here, the fact that the property exists gets kind of hidden behind the mixin.

How would you feel about commenting it out and putting in a note?
```ts
// id?: FHIRId; // Provided by HasId mixin
```

Reply:
Similar goes for `name`.

Reply:
And... for all the other types that use these mixins.